src/EnergyPlus/OutsideEnergySources.cc

```diff
@@ -331,23 +367,55 @@ void OutsideEnergySourceSpecs::size(EnergyPlusData &state)
     std::string typeName;
     if (this->EnergyType == DataPlant::PlantEquipmentType::PurchChilledWater) {
         typeName = "Cooling";
-    } else { // Heating
+    } else if (this->EnergyType == DataPlant::PlantEquipmentType::PurchHotWater) {
```

Comment:
This `if-else` could be a lookup in a `constexpr std::array<std::string_view>`, no?

Reply:
Also, just create the concatenated `"SizeDistrict" + typeName` string once here, and then reuse.

Reply:
As @amirroth suggested long ago, `typename` should be the actual object type, e.g. "DistrictHeating:Water" etc.
    `std::string_view typeName = DataPlant::PlantEquipTypeNames[static_cast<int>(this->EnergyType)];`

This replaces the entire if/elseif block. And everywhere that `typeName` is used, remove "District" from the format strings. @Myoldmopar I can make these changes and push if you want.

This should fix the eio component sizing outputs which are currently like:
```
Component Sizing Information, DistrictHeatingWater, PURCHASED HEATING, User-Specified Nominal Capacity [W], 1000000.00000
```
to be
```
Component Sizing Information, DistrictHeating:Water, PURCHASED HEATING, User-Specified Nominal Capacity [W], 1000000.00000
```

Reply:
@mjwitte that would be fine.  As of now I don't have any changes to the C++ code, just transition changes.  So if you change this it wouldn't conflict and would only clean up my diffs.

Reply:
Done, but this won't reduce the diffs, it'll just make them correct.

Reply:
Ahhh that's fine anyway.  Thanks